### DISCUSSION TITLE
Problem booting from -flat.vmdk image with partitions

### DISCUSSION THREAD

**Discussion by jonsito** [ltsp/ltsp#851] [General]
### Discussed in https://github.com/ltsp/ltsp/discussions/849

<div type='discussions-op-text'>

<sup>Originally posted by **jonsito** August  2, 2023</sup>
Hello all,

I'm using Virtualbox to generate client images. It works fine either via NBD or NFS with squashfs generated images from flat.vmdk virtualbox file

Aside note: NFS performance is by far large better than NBD when working with multiple simultaneous clients and large squashfs image files. 

Surfing in code I've found ( file 55-images.sh in function mount_image_src() ) some coments saying that it is possible to export and use directly vmdk image without conversion to squashfs

My vbox image is properly flat.vmdk formatted with /  (ext4) as partition #1 and /boot/efi (vfat) as #2. 

My grub file says as attached image
Client starts, but fails trying to mount partition #1 from vmdk image (as shown in attached image). It seems as ltsp code does not know how to handle partitions on vmdk image and makes 'losetup' fail... perhaps a missing "-P" or "--partscan" in losetup command?

What am I doing wrong? Any ideas?
Thanks in advance
![Captura de pantalla de 2023-08-02 11-41-16](https://github.com/ltsp/ltsp/assets/828257/12e2633f-7c7a-48d3-83e5-66a464f02f7c)

![Captura de pantalla de 2023-08-02 11-48-47](https://github.com/ltsp/ltsp/assets/828257/56dc45da-bb84-4122-a26f-7041278d7ae0)
</div>

**Comment by jonsito** [comment]
More issues:
- If I extract partition 1 from vmdk image and try to nfsmount and use on client, everything goes fine. No need to create squashfs.

- losetup cmd included in initrd comes from klibc and does not handle partitions. Don't know how difficult is include "standard" losetup in initrd instead of klibc version one?

**Comment by alkisg** [comment]
Are you using a custom command line?
Do you include the `loop.max_part=9` parameter that ltsp.ipxe has?

**Comment by jonsito** [comment]
Wow! it rocks.
Thank you very much. 

Performance is impressive when compared againts squashfs'd Ubuntu 20.04 rootfs.

Yes: I'm using my own grub.cfg file: we handle an students lab at the Universidad Politécnica de Madrid, and our client pc's have multiple startup configurations.

Again: thank you so much for your help.
Juan Antonio
